main.js

```diff
@@ -0,0 +1,156 @@
+        origin: 'Ireland',
+        imageUrl: 'https://goo.gl/YNb74O'
+      },
+      {
+        itemId: 8,
+        name: 'TR-09',
+        brand: 'Roland',
+        price: 399,
+        description: 'A drum machine for people with no desk space.',
+        details: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
+        origin: 'Japan',
+        imageUrl: 'https://goo.gl/gJYUxz'
+      }
+    ]
+  },
+  details: {
+    item: null
+  }
+}
+
+function createCard(item) {
+  var $card = document.createElement('div')
+  $card.classList.add('card')
+
+  var $image = document.createElement('img')
+  $card.appendChild($image)
+  $image.classList.add('card-img-top')
+  $image.setAttribute('src', item.imageUrl)
+  $image.classList.add('images')
+
+  var $cardBody = document.createElement('div')
+  $card.appendChild($cardBody)
+  $cardBody.classList.add('card-body')
+  $card.style.height = '500px'
+
+  var $brand = document.createElement('h5')
+  $cardBody.appendChild($brand)
+  $brand.classList.add('card-title')
+  $brand.textContent = item.brand
+
+  var $name = document.createElement('h5')
+  $cardBody.appendChild($name)
+  $name.classList.add('card-title')
+  $name.textContent = item.name
+
+  var $price = document.createElement('h5')
+  $cardBody.appendChild($price)
+  $price.classList.add('card-title')
+  $price.textContent = item.price
+
+  return $card
+}
+
+function createGridAndHeading(allItems) {
+  var $grid = document.createElement('div')
+  $grid.classList.add('container')
+
+  var $header = document.createElement('h1')
+  $grid.appendChild($header)
+  $header.textContent = 'Jamazon'
+  $header.classList.add('jamazonText')
+
+  var $row1 = document.createElement('div')
+  $row1.classList.add('row')
+  $grid.appendChild($row1)
+
+  // loop through allItems, create a div, render the card and pass it an item, append
+  // the div to the grid. append the card into the div
+  for (var i = 0; i < allItems.length; i++) {
+    var $cardDiv = document.createElement('div')
+    $cardDiv.classList.add('col-md-3')
+    $row1.appendChild($cardDiv)
+    var $newCard = createCard(allItems[i])
+    $cardDiv.appendChild($newCard)
+  }
+  return $grid
+}
+// Define a function that renders the entire app state and inserts it into the view.
+function renderGridAndHeader() {
+  var i = createGridAndHeading(app.catalog.items)
+  document.body.appendChild(i)
```

Comment:
Instead of appending this to the `body`, there is a specific place to insert the rendered catalog in `index.html`.

Reply:
Done.  

function renderGridAndHeader() {
  var i = createGridAndHeading(app.catalog.items)
  var $catalog = document.querySelector("div[data-view='catalog']")
  $catalog.appendChild(i)
}
renderGridAndHeader()